src/Core/Module/ModuleEloquentException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Module;
+
+/**
+ * Module Exception whose message can be displayed directly to end-user
+ */
+class ModuleEloquentException extends ModuleException
```

Comment:
Naming can be improved

Reply:
`ModuleDisplayedException` ? `ModuleAlertException` ?

I still think an interface could be nice for this use case, something like:

```php
interface DisplayedExceptionInterface
{
    /**
     * @return array
     */
    public function getDisplayedMessage();
}
```
or
```php
interface AlertExceptionInterface
{
    /**
     * @return array
     */
    public function getAlertMessages();
}
```
(not sure if a string or an array of string is best)

And it doesn't prevent `ModuleEloquentException` to be created and implement this interface.
I see several advantages to using an interface:
- modules can create their own exception without extending a core one
- we can use this behaviour even for core exception later, so we don't have to implement their management several times


Reply:
Actually, now that you say this, I'm thinking: maybe using the interface we can get rid of the hook ?

The hook was there to tell to the core "eyh, please consider these exceptions as EloquentExceptions (or whatever name we choose) although they do not extend it"
But the interface would do it nicely. And it's a standard practice, for example people creating new Loggers use the Monolog interface.

Reply:
I think it's also easier for developers, just need to add the interface and it's done No hook registering And the core code stays cleaner (no hook to call in between)

Reply:
What about:
- `ModuleErrorMessageException` (easy to understand)
- `ModuleErrorException` (maybe more semantic alternative)

Not a fan of "Eloquent", sorry 😛 

In most cases, developers will just throw a `ModuleErrorMessageException` (or a child class) and get done with it, no need for the hook in the end.

Reply:
I prefer the shorter `ModuleErrorException` personnaly